src/Sylius/Bundle/ResourceBundle/Controller/DomainManager.php

```diff
@@ -84,7 +86,7 @@ public function create($resource)
             $this->flashHelper->setFlash('success', 'create');
         }
 
-        $this->dispatchEvent('post_create', new ResourceEvent($resource));
+        $this->dispatchEvent(sprintf('post_%s', $eventName ?: 'create'), new ResourceEvent($resource));
```

Comment:
and you should remove that condition and use the default value described previously.